### DISCUSSION TITLE
API Documentation

### DISCUSSION THREAD

**Discussion by kentris1123** [bnpr/Malt#87] [Feedback & Feature Requests]
Hi!

Could you add an API Documentation for built-in functions such as get_diffuse_half?

Malt is such a great plugin that I can make whole pipeline within an hour after download.
But I'm stumbled when I was to create something other than what Shader-Examples covers.
It'll be fantastic if we don't have to be guessing all the API does.

Cheers!

**Comment by pragma37** [comment]
Yes, this is on my to-do list.
I'll start working on it after I complete the initial nodes implementation.

Meanwhile, feel free to ask any specific questions you have.

**Comment by Pacomatic1** [comment]
Came here to say the same thing, there's no documentation whatsoevver when it comes to making a new pipeline from scratch.
I'vve basically been forced to copy the MiniPipeline exxample and then 'figure it out' by reading the NPR Pipeline and the Malt source code, but that has so far been misery.

Plus, once you add some docs, I think this thing will become so much more useful. Right now, people only really praise it and consider it for its usage in NPR rendering, but that's because the NPR pipeline is, to my knowledge, the only pipeline that exxists. That's quite disappointing, considerig the fact that this thing is so ridiculously exxtensible thanks to its custom pipeline capabilities.
But that is really hard right now, so nobody has done it.

As I like to say: A good API is nothing without good documentation.
